paxml/learners.py

```diff
@@ -588,8 +631,16 @@ def scale_gradients_by_optimizer(
   ) -> Tuple[NestedMap, JTensor]:
     optimizer_mask, default_mask = self.get_masks(var_weight_hparams)
 
-    all_grads, all_valid_step = self.scale_gradients(
-        jax.tree_map(lambda x, y: x * y, raw_grads, default_mask),
+    raw_grads = jax.tree_map(lambda x, y: x * y, raw_grads, default_mask)
```

Comment:
let's not reuse `raw_grads`, let's call this `grads_after_mask` because you've introduced a subtle bug here if you look at line line 659 inside the auxiliary_optimizers loop, you are now combining this outer mask with inner mask

i would not overwrite `raw_grads` variable, just 

```
grads_after_mask = jax.tree_map(lambda x, y: x * y, raw_grads, default_mask)
grad_norm, all_valid_step = self.get_grad_norm_valid_step(
        grads_after_mask,
        optimizer_name='main',
    )
```

so that inside auxiliary_optimizers loop, `raw_grads` is only added to each auxiliary optimizer mask